docs/advanced/pyproject_toml.md

```diff
@@ -0,0 +1,154 @@
+# Pyproject.toml in pixi
+We support the use of the `pyproject.toml` as our manifest file in pixi.
+This allows the user to keep one file with all configuration.
+The `pyproject.toml` file is a standard for Python projects.
+We don't advice to use the `pyproject.toml` file for anything else than python projects, the `pixi.toml` is better suited for other types of projects.
+
+## Initial setup of the `pyproject.toml` file
+When you already have a `pyproject.toml` file in your project, you can add the following section to it:
+```toml
+[tool.pixi.project]
+name = "my_project"
+channels = ["conda-forge"]
+platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
+```
+This is the minimum requirement for pixi to understand and parse the project.
+
+If you use `pixi init` in a folder that has a `pyproject.toml` file, pixi will automatically add the `[tool.pixi.project]` section to the file.
+And will also add some defaults to the `.gitignore` and `.gitattributes` file.
+
+## Python dependency
+The `pyproject.toml` file supports the `requires_python` field.
+Pixi understands that field and automatically adds the version to the dependencies.
+
+This is an example of a `pyproject.toml` file with the `requires_python` field, which will be used as the python dependency:
+```toml title="pyproject.toml"
+[project]
+name = "my_project"
+requires-python = ">=3.9"
+
+[tool.pixi.project]
+name = "my_project"
+channels = ["conda-forge"]
+platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
+```
+Which is equivalent to:
+```toml title="equivalent pixi.toml"
+[project]
+name = "my_project"
+channels = ["conda-forge"]
+platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
+
+[dependencies]
+python = ">=3.9"
+```
+
+## Dependency section
+The `pyproject.toml` file supports the `dependencies` field.
+Pixi understands that field and automatically adds the dependencies to the project as `[pypi-dependencies]`.
+
+This is an example of a `pyproject.toml` file with the `dependencies` field:
+```toml title="pyproject.toml"
+[project]
+name = "my_project"
+requires-python = ">=3.9"
+dependencies = [
+    "numpy",
+    "pandas",
+    "matplotlib",
+]
+
+[tool.pixi.project]
+name = "my_project"
+channels = ["conda-forge"]
+platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
+```
+
+Which is equivalent to:
+```toml title="equivalent pixi.toml"
+[project]
+name = "my_project"
+channels = ["conda-forge"]
+platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
+
+[pypi-dependencies]
+numpy = "*"
+pandas = "*"
+matplotlib = "*"
+
+[dependencies]
+python = ">=3.9"
+```
+
+You can overwrite these with conda dependencies by adding them to the `dependencies` field:
```

Comment:
I do not think this is the case anymore, as I pulled out all the bits related to conda to pypi mapping (which is required to detect when a conda dependency refers to the same package as a pypi dependency).

I can bring that back if you want?

Reply:
I would be in favor of having this behavior as imo conda packages are more predictable than pypi packages.
Alternatively, we could add something like
```toml
[tool.pixi]
ignore-pyproject-toml-dependencies = true
```

Reply:
Currently, in pixi if you add the dependency to the conda dependencies explicitly. E.g you use 'flask' as pypi dependency and 'click' as conda, then the pypi flask which depends on click will use the conda dependency.

Reply:
Hum. So maybe this is still done automatically by pixi then. I’ll check tomorrow.

Reply:
Yeah this is just default pixi behavior. We always ignore pypi dependencies which are already installed in your environment.

Pixi's internal flow:
Install conda environment -> check which pypi-dependencies it still needs to install using a pypi-conda map -> install missing pypi packages.

This is all hidden in the installation logic. Which is related to my reaction in https://github.com/prefix-dev/pixi/issues/532#issuecomment-2022175953